src/sys/socket/ffi.rs

```diff
@@ -37,7 +37,7 @@ pub struct cmsghdr {
     pub cmsg_len: type_of_cmsg_len,
     pub cmsg_level: c_int,
     pub cmsg_type: c_int,
-    pub cmsg_data: [usize; 0]
+    pub cmsg_data: [size_t; 0]
```

Comment:
`type_of_cmsg_len` for `cmsg_data` was totally wrong.  If you look at `CMSG_DATA` on Linux, it basically says that the data field begins at a location suitable for an aligned `struct cmsghdr`.  On FreeBSD, `CMSG_DATA` basically says that the data field should begin at a location suitably aligned for a `register_t`, which is either a `__int32_t` or an `__int64_t` depending on the platform.